### PR TITLE
feature(posts): responsiveness

### LINKED ISSUE #85: Feature: add responsiveness for the posts view
### Type of feature
- implement responsiveness for the posts view 

- responsiveness has not yet been covered
- there are no media queries throughout the entire app therefore if any new work is being done related to responsiveness this is a feature, since there is no correction of existing code or implementation

## Prerequisites
- when working on responsiveness we must consider the entire screen, which includes the header and the footer
- we can either work on the entire screen view in one issue, or if it is split into smaller issues - header and footer's responsiveness should be added first

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by hossameldinfahmi**
## This PR Closes Issue https://github.com/AnguHashBlog/angular/issues/85
closes #85 

## What type of PR is this? (check all applicable)

- [x] 🅰️ Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Mobile & Desktop Screenshots/Recordings
[Attach screenshots or recordings if applicable]
Screenshot 2024-03-18 021819

## Added to documentation?
- [ ] 📜 README.md
- [x] 🙅 no documentation needed

## [Optional] Post-deployment tasks
[Specify any post-deployment tasks that need to be performed]

## [Optional] What gif best describes this PR or how it makes you feel? 
[Embed gif or describe the feeling in plain text]

### CODE REVIEW

File: angular-app/src/app/components/posts/posts.component.scss [hunk 1]
```diff
@@ -47,6 +47,16 @@
         }
       }
     }
+
+      @media (max-width: 768px) {
```
Comment (monacodelisa) [inline angular-app/src/app/components/posts/posts.component.scss:51]: @hossameldinfahmi this is a good start, please adjust the placement of the media query - it should be placed below all the other styles.

also - in your screenshot the header and the footer are not present - have you checked to see how they look ? Responsiveness is something that relates to the whole view and before we jump to making adjustments to the inner component, adjustments should be made to the outer components that wrap up the view. 

What you can do is - ither handle header and footer within this issue, or if you really only want to work on the post cards - you can wait until the responsiveness has been applied to the header and the footer. 

Please let me know what you prefer to do and if you need any help, maybe I can assign the header & footer responsiveness to someone and we can help you get to the part that you would like to work on.

**Review by monacodelisa — CHANGES_REQUESTED** [review]

File: angular-app/src/app/components/posts/posts.component.scss (hunk 1 above)
Comment (hossameldinfahmi) [reply angular-app/src/app/components/posts/posts.component.scss:51]: @monacodelisa Thanks for your notice, that's great i will work on the header and the footer responsive and I will also resolve your comments.

**Comment by hossameldinfahmi** [comment]
@monacodelisa , I've improved the responsiveness of the header and I moved the relevant styles to the bottom of the SCSS page. The footer's responsiveness is already satisfactory and doesn't require any further modifications.

File: angular-app/src/app/components/posts/posts.component.scss (hunk 1 above)
Comment (monacodelisa) [reply angular-app/src/app/components/posts/posts.component.scss]: @hossameldinfahmi cool, looks good carry on 👍

**Comment by monacodelisa** [comment]
> @monacodelisa , I've improved the responsiveness of the header and I moved the relevant styles to the bottom of the SCSS page. The footer's responsiveness is already satisfactory and doesn't require any further modifications.

ok will have a look at your changes

File: angular-app/src/app/components/header/header.component.scss [hunk 2]
```diff
@@ -93,4 +93,24 @@
...
+        align-items: center;
+        align-items: center;
+      }
+      .series{
+        flex-direction: column;
+        align-items: center;
+      }
+    }
+  }
 }
```
Comment (monacodelisa) [inline angular-app/src/app/components/header/header.component.scss:116]: @hossameldinfahmi great job on this it's looking very nice I just had a look, one final thing and I can approve it for merge right after - do you see that there is one `}` left here below ? 

the entire media query 

```
@media (max-width: 767px) {
    .toolbar-row {
      flex-direction: column;
      padding-bottom: 12px;

      .toolbar-row-start,
      .toolbar-row-end {
        width: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
        align-items: center;
      }
      .series{
        flex-direction: column;
        align-items: center;
      }
    }
  }
```
needs to go below the last `}` 
there will be more styles probably added in the future and it is a very good start to making the app responsive :)

Comment (hossameldinfahmi) [reply angular-app/src/app/components/header/header.component.scss:117]: @monacodelisa Thanks a lot i moved it.

**Review by monacodelisa — APPROVED** [review]

**Review by cleancodecraft — APPROVED** [review]

**Review by samplelittlered — APPROVED** [review]

**Comment by monacodelisa** [comment]
@hossameldinfahmi thank you :)

**Pull request merged**